### ISSUE TITLE
does not observe `vendor-dir` config in composer.json

### ISSUE THREAD

**Issue by turadg** [hhvm/hhast#119]
All the `bin` scripts expect to find `hh_autoload.php` within `vendor` directory: https://github.com/hhvm/hhast/search?q=autoload.php&unscoped_q=hh_autoload.php

While `vendor` is the default vendor directory, Composer allows customizing that with the `vendor-dir` property: https://getcomposer.org/doc/06-config.md#vendor-dir

**Comment by fredemmott** [comment]
yeah, this is currently expected by all our open source projects - for example, hhvm-autoload also won't work unless this is the case.

**Comment by turadg** [comment]
We've [forked `hhvm-autoload`](https://github.com/hhvm/hhvm-autoload/compare/master...quizlet:karoun/root-dir-legacy) so we can have `vendor/composer` instead of `vendor`. I'd like to get us to mainstream so don't want to fork hhast too.

Sounds like you'd accept a PR to read the config. I'm exploring options and will come back to this if that's the straightest path.

**Comment by lexidor** [comment]
@turadg @fredemmott I will add a `find_autoloader.hack` file to the `/bin` scripts that `hhvm-autoload` exports. This will allow all `/bin` scripts to do this

```HACK
require_once __DIR__ . '/find_autoloader.hack';
Facebook\AutoloadMap\dynamic_initialize();
```

instead of needing to write their own implementation of the climb-until-you-find-a-vendor-dir script over and over.

That will allow us to make patches to support non-standard vendor-dir and such in one place.

**Comment by lexidor** [comment]
My plan was misguided. `vendor/bin/hhast-lint.hack` and friends are symlinks.
They can not require neighboring scripts.
In order for this approach to work, we would need to embed dynamic_initialize into hhvm.
Would it be worth it still?
